Source/WebCore/dom/make_names.pl

```diff
@@ -759,145 +822,378 @@ END
+            print F "        return TagName::$allTags{$tagKey}{tagEnumValue};\n";
+        }
+        print F "    default:\n";
+        print F "        return tagName;\n";
+        print F "    }\n";
+        print F "}\n";
+        print F "\n";
+    }
+    print F "} // namespace WebCore\n";
+    close F;
+}
+
+sub findMaxStringLength
 {
-    my $allTags = shift;
+    my $candidates = shift;
 
     my $maxLength = 0;
-    foreach my $tagName (keys %{$allTags}) {
-        my $tagLength = length($tagName);
-        $maxLength = $tagLength if $tagLength > $maxLength;
+    for my $candidate (@$candidates) {
+        $maxLength = length($candidate->{string}) if length($candidate->{string}) > $maxLength;
     }
+
     return $maxLength;
 }
 
-sub tagsWithLength
+sub candidatesWithStringLength
 {
-    my $allAttrs = shift;
+    my $candidates = shift;
     my $expectedLength = shift;
 
-    my @tags = (); 
-    foreach my $tagName (sort keys %{$allAttrs}) {
-        push(@tags, $tagName) if length($tagName) == $expectedLength;
-    }
-    return @tags;
+    return grep { length($_->{string}) == $expectedLength } @$candidates;
 }
 
-sub generateFindTagForLength
+sub generateFindNameForLength
 {
     my $indent = shift;
-    my $tagsRef = shift;
+    my $candidates = shift;
     my $length = shift;
     my $currentIndex = shift;
+    my $enumClass = shift;
 
-    my @tags = @{$tagsRef};
-    my $tagCount = @tags;
-    if ($tagCount == 1) {
-        my $tag = $tags[0];
+    my $candidateCount = @$candidates;
+    if ($candidateCount == 1) {
+        my $candidate = $candidates->[0];
+        my $string = $candidate->{string};
+        my $enumValue = $candidate->{enumValue};
         my $needsIfCheck = $currentIndex < $length;
         if ($needsIfCheck) {
             my $lengthToCompare = $length - $currentIndex;
             if ($lengthToCompare == 1) {
-                my $letter = substr($tag, $currentIndex, 1);
+                my $letter = substr($string, $currentIndex, 1);
                 print F "${indent}if (buffer[$currentIndex] == '$letter') {\n";
             } else {
                 my $bufferStart = $currentIndex > 0 ? "buffer.data() + $currentIndex" : "buffer.data()";
-                print F "${indent}static constexpr UChar ${tag}Rest[] = { ";
+                print F "${indent}static constexpr characterType rest[] = { ";
                 for (my $index = $currentIndex; $index < $length; $index = $index + 1) {
-                    my $letter = substr($tag, $index, 1);
+                    my $letter = substr($string, $index, 1);
                     print F "'$letter', ";
                 }
                 print F "};\n";
-                print F "${indent}if (!memcmp($bufferStart, ${tag}Rest, $lengthToCompare * sizeof(UChar))) {\n";
+                print F "${indent}if (!memcmp($bufferStart, rest, $lengthToCompare * sizeof(characterType))) {\n";
```

Comment:
We shouldn't have curly braces around a single line statement.

Reply:
Filed https://bugs.webkit.org/show_bug.cgi?id=244524 for this.